webtransport/csp-fail.https.window.js

```diff
@@ -14,6 +14,18 @@ promise_test(async t => {
   document.head.appendChild(meta);
 
   let wt = new WebTransport(webtransport_url('custom-response.py?:status=200'));
-  await promise_rejects_dom(t, 'SecurityError', wt.ready, 'ready promise should be rejected');
-  await promise_rejects_dom(t, 'SecurityError', wt.closed, 'closed promise should be rejected');
+
+  // Sadly we cannot use promise_rejects_dom as the error constructor is
+  // WebTransportError rather than DOMException.
+  for (const name of ["ready", "closed"]) {
+    try {
+      await wt[name];
+      test.unreached_func(`${name}: should have rejected promise`);
+    } catch (e) {
+      assert_true(e instanceof WebTransportError);
+      assert_equals(e.name, 'WebTransportError', `${name}: WebTransportError`);
+      assert_equals(e.source, 'session', `${name}: source`);
+      assert_equals(e.streamErrorCode, null, `${name}: streamErrorCode`);
+    }
+  }
```

Comment:
Maybe something like this could work, similar to some of the other WebTransport WPTs checking for WebTransportErrors?
```suggestion
  const e = await wt.ready.catch(e => e);
  await promise_rejects_exactly(t, e, wt.ready, 'ready promise should be rejected');
  await promise_rejects_exactly(t, e, wt.closed, 'closed promise should be rejected');
  assert_true(e instanceof WebTransportError);
  assert_equals(e.name, 'WebTransportError', 'WebTransportError');
  assert_equals(e.source, 'session', 'source');
  assert_equals(e.streamErrorCode, null, 'streamErrorCode');
```

Reply:
+1